scripts/g.extension/testsuite/test_addons_download.py

```diff
@@ -0,0 +1,141 @@
+"""
+TEST:      test_addons_download.py
+
+AUTHOR(S): Stefan Blumentrath <stefan dot blumentrath at nina dot no)
+           based on test_addons_modules.py by
+           Vaclav Petras <wenzeslaus gmail com>
+
+PURPOSE:   Test for g.extension individual modules/extensions download
+
+COPYRIGHT: (C) 2020 Stefan Blumentrath, Vaclav Petras,
+           and by the GRASS Development Team
+
+           This program is free software under the GNU General Public
+           License (>=v2). Read the file COPYING that comes with GRASS
+           for details.
+"""
+
+import sys
+import unittest
+
+from pathlib import Path
+
+from grass.gunittest.case import TestCase
+from grass.gunittest.main import test
+from grass.gunittest.utils import silent_rmtree
+
+ms_windows = sys.platform == "win32"
+
+
+class TestModuleDownloadFromDifferentSources(TestCase):
+
+    # MS Windows install function requires absolute paths
+    install_prefix = Path("gextension_test_install_path").absolute()
+
+    files = [
+        install_prefix.joinpath("scripts", "r.example.plus"),
+        install_prefix.joinpath("docs", "html", "r.example.plus.html"),
+    ]
+
+    def setUp(self):
+        """Make sure we are not dealing with some old files"""
+        if self.install_prefix.exists():
+            files = list(path.name for path in self.install_prefix.iterdir())
+            if files:
+                RuntimeError(
+                    "Install prefix path '{}' contains files {}".format(
+                        str(self.install_prefix), files
+                    )
+                )
+
+    def tearDown(self):
+        """Remove created files"""
+        silent_rmtree(str(self.install_prefix))
+
+    @unittest.skipIf(ms_windows, "currently not supported on MS Windows")
+    def test_github_install(self):
+        """Test installing extension from github"""
+
+        self.assertModule(
+            "g.extension",
+            extension="r.example.plus",
+            url="https://github.com/wenzeslaus/r.example.plus",
+            prefix=str(self.install_prefix),
+        )
+
+        # Modules with non-standard branch would be good for testing...
+        self.assertModule(
+            "g.extension",
+            extension="r.example.plus",
+            url="https://github.com/wenzeslaus/r.example.plus",
+            prefix=str(self.install_prefix),
+        )
```

Comment:
Is this a placeholder for a future test?

Reply:
In a way, but as it is now it is only a duplication. So I removed it in [b0e1215](https://github.com/OSGeo/grass/pull/1158/commits/b0e12150274791f3dd7f03c62a47245095145a15)